tests/integration/common/s2n_test_scenario.py

```diff
@@ -108,6 +109,43 @@ def get_libcrypto():
     return str(os.getenv("S2N_LIBCRYPTO")).strip('"')
 
 
+def get_libcrypto_match(s2nx="../../bin/s2nc"):
```

Comment:
What decisions can we make from this? My first thought would be to make this check in a CI shellscript before running the integ tests, and fail if there is a mismatch.

Even though libre and boring (and I assume awslc) make it difficult to determine true identity, could we look at an environment variable, and then check for !other_provider? Like, `if S2N_LIBCRYPTO == boring ; verify !openssl`?

If we could make the check in a shell script we could also use it across the v2 library. There is added complexity there because we could compile with OpenSSL, but verify against the BoringSSL provider.

But all that may be beyond the scope of this pr, please let me know.

Reply:
The ldd output is just easier to deal with in python.  The goal was to get this link check done in the same environment space as the integration test, to capture any polluted LD_LIBRARY_PATH and boost confidence in the output that the correct LibCrypto was used.